python/lsst/daf/butler/_butler.py

```diff
@@ -2235,6 +2241,27 @@ def transfer_from(
         else:
             log.log(VERBOSE, "All required dataset types are known to the target Butler")
 
+        dimension_records: Dict[DimensionElement, Dict[DataCoordinate, DimensionRecord]] = defaultdict(dict)
+        if transfer_dimension_data:
+            # Collect all the dimension records for these refs.
+            # All dimensions are to be copied but querying this butler
+            # universe.
```

Comment:
Something wrong with this sentence.

Reply:
I think I'm trying to say that even though I'm copying records from the source butler I'm limiting the dimensions to those understood by the destination butler...